### PR TITLE
added libogg, first step toward building ffmpeg with libvpx

### PR DESCRIPTION
**Pull request by zoeith**
See https://github.com/conda-forge/ffmpeg-feedstock/issues/8 for motivation.

### CODE REVIEW

File: recipes/libogg/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+source:
+    fn: libogg-{{ version }}.tar.gz
+    url: http://downloads.xiph.org/releases/ogg/libogg-{{ version }}.tar.gz
+    # md5 is OK but sha256 is desirable.
+    sha256: e19ee34711d7af328cb26287f4137e70630e7261b17cbe3cd41011d73a654692
+
+build:
+    number: 0
+    # If the installation is complex, or different between Unix and Windows, use separate bld.bat and build.sh files instead of this key.
+    script: true
```
Comment (patricksnape) [inline recipes/libogg/meta.yaml]: You can remove this. But (given the script) we likely want `skip: True  # [win]`

File: recipes/libogg/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        - gcc
+
+#no test for library? :-/
+test:
+    test -f ${PREFIX}/lib/libogg.a
+
+
+about:
+    home: https://www.xiph.org/ogg/doc/
+    # Remember to specify the license variants for BSD, Apache, GPL, and LGLP.
```
Comment (patricksnape) [inline recipes/libogg/meta.yaml]: You can remove this boilerplate.

File: recipes/libogg/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    # If the installation is complex, or different between Unix and Windows, use separate bld.bat and build.sh files instead of this key.
+    script: true
+
+requirements:
+    build:
+        - gcc
+
+#no test for library? :-/
+test:
+    test -f ${PREFIX}/lib/libogg.a
```
Comment (patricksnape) [inline recipes/libogg/meta.yaml]: Needs the 'bullet point' -> `- test -f ${PREFIX}/lib/libogg.a`

File: recipes/libogg/build.sh [hunk 4]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+./configure \
+        --prefix="${PREFIX}"
+
+make
```
Comment (jakirkham) [inline recipes/libogg/build.sh]: `make check`?

File: recipes/libogg/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    url: http://downloads.xiph.org/releases/ogg/libogg-{{ version }}.tar.gz
+    sha256: e19ee34711d7af328cb26287f4137e70630e7261b17cbe3cd41011d73a654692
+
+build:
+    number: 0
+    skip: True  # [win]
+
+requirements:
+    build:
+        - gcc
```
Comment (jakirkham) [inline recipes/libogg/meta.yaml]: Yeah, we are going to have to fix this. I'm starting on the `ffmpeg` stack.

File: recipes/libogg/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+test:
+    commands:
+        - test -f ${PREFIX}/lib/libogg.a
+
+
+about:
+    home: https://www.xiph.org/ogg/doc/
```
Comment (jakirkham) [inline recipes/libogg/meta.yaml]: http://xiph.org/ogg/

File: recipes/libogg/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    license: BSD-Style
```
Comment (jakirkham) [inline recipes/libogg/meta.yaml]: BSD 3-Clause

File: recipes/libogg/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    number: 0
+    skip: True  # [win]
+
+requirements:
+    build:
+        - gcc
+
+test:
+    commands:
+        - test -f ${PREFIX}/lib/libogg.a
```
Comment (jakirkham) [inline recipes/libogg/meta.yaml]: Tests for dynamic libraries?

File: recipes/libogg/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+        - test -f ${PREFIX}/lib/libogg.la
+        - test -f ${PREFIX}/lib/libogg.dylib
```
Comment (jakirkham) [inline recipes/libogg/meta.yaml]: We might need a selector as Linux and OS X use different extensions. This should work.

```
- test -f ${PREFIX}/lib/libogg.so     # [linux]
- test -f ${PREFIX}/lib/libogg.dylib  # [osx]
```

File: recipes/libogg/build.sh (hunk 4 above)
Comment (jakirkham) [inline recipes/libogg/build.sh]: Done below.

File: recipes/libogg/meta.yaml (hunk 8 above)
Comment (jakirkham) [inline recipes/libogg/meta.yaml]: Done below.

**Comment by 183amir** [comment]
Does this work without listing `gcc`?
You can add me as one the maintainers.

**Pull request merged**